packages/manager/modules/overthebox/src/overthebox/autoconfigure/dns/autoconfigure-dns.controller.js

```diff
@@ -0,0 +1,240 @@
+          'overTheBox_autoconfigure_loading_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  loadNameserver() {
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dns/nameserver`)
+      .then((response) => {
+        this.nameServers = response.data;
+        return response;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_loading_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  addNameserver() {
+    this.nameserver.server = '';
+    this.nameserver.priority = '';
+    this.showAddServer = true;
+    this.displayAddServerButton = false;
+  }
+
+  createNameServer() {
+    this.errorMessage = '';
+    this.successMessage = '';
+
+    const params = {
+      configuration: {
+        server: this.nameserver.server,
+      },
+      priority: this.nameserver.priority,
+    };
+
+    return this.$http
+      .post(
+        `/overTheBox/${this.serviceName}/configuration/dns/nameserver`,
+        params,
+      )
+      .then(() => {
+        // Display success message
+        this.successMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_dns_server_create_succeed',
+        );
+
+        // Reload nameserver table
+        this.showAddServer = false;
+        this.displayAddServerButton = true;
+        return this.loadNameserver();
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_create_failed',
+          { errorMessage: error.data.message },
+        );
+      });
+  }
+
+  cancelCreateNameServer() {
+    this.showAddServer = false;
+    this.displayAddServerButton = true;
+  }
+
+  removeNameserver(row) {
+    this.errorMessage = '';
+    this.successMessage = '';
+    return this.$http
+      .delete(
+        `/overTheBox/${this.serviceName}/configuration/dns/nameserver/${row.id}`,
+      )
+      .then(() => {
+        // Display success message
+        this.successMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_dns_server_remove_succeed',
+        );
+
+        // Reload nameserver table
+        return this.loadNameserver();
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_remove_failed',
+          { errorMessage: error.data.message },
+        );
+      });
+  }
+
+  addLocalDomain() {
+    this.localDomain.hostname = '';
+    this.localDomain.ip = '';
+    this.localDomain.priority = '';
+    this.showAddLocalDomain = true;
+    this.displayAddLocalDomainButton = false;
+  }
+
+  createLocalDomain() {
+    this.errorMessage = '';
+    this.successMessage = '';
```

Comment:
This combination is repeated on others place, you have to factorize.

```suggestion
resetMessages() {
    this.errorMessage = '';
    this.successMessage = '';
}

this.resetMessages();
```